### PR TITLE
Add created annotation to returned values

### PR DESCRIPTION
**Pull request by syou6162**

### CODE REVIEW

File: graph_annotation.go [hunk 1]
```diff
@@ -20,10 +20,21 @@ type GraphAnnotation struct {
...
+		return nil, err
+	}
+
+	var createdAnnotation GraphAnnotation
+
+	err = json.NewDecoder(resp.Body).Decode(&createdAnnotation)
+	if err != nil {
+		return nil, err
+	}
+	return &createdAnnotation, err
```
Comment (haya14busa) [inline graph_annotation.go]: [nit] err is always nil, so return "nil" explicitly.

File: graph_annotation.go [hunk 2]
```diff
@@ -20,10 +20,21 @@ type GraphAnnotation struct {
...
 	resp, err := c.PostJSON("/api/v0/graph-annotations", annotation)
 	defer closeResponse(resp)
-	return err
+
+	if err != nil {
+		return nil, err
+	}
+
+	var createdAnnotation GraphAnnotation
+
```
Comment (haya14busa) [inline graph_annotation.go]: [nit] I prefer to remove this blank line.

File: graph_annotation.go [hunk 3]
```diff
@@ -20,10 +20,21 @@ type GraphAnnotation struct {
...
+func (c *Client) CreateGraphAnnotation(annotation *GraphAnnotation) (*GraphAnnotation, error) {
 	resp, err := c.PostJSON("/api/v0/graph-annotations", annotation)
 	defer closeResponse(resp)
-	return err
+
+	if err != nil {
+		return nil, err
+	}
+
+	var createdAnnotation GraphAnnotation
```
Comment (haya14busa) [inline graph_annotation.go]: [nit]
In Go, shorter name is preferred especially if the lifetime is short.
I think 'annotation' or something is enough in this case.  but you can ignore this, it's ok as it is.

File: graph_annotation_test.go [hunk 4]
```diff
@@ -73,6 +78,30 @@ func TestCreateGraphAnnotation(t *testing.T) {
 	if err != nil {
 		t.Error("err shoud be nil but: ", err)
 	}
+
+	if annotation.Service != "My-Blog" {
+		t.Error("request sends json including Service but: ", annotation.Service)
+	}
```
Comment (haya14busa) [inline graph_annotation_test.go]: [nit]
I think the error message is not helpful. ( what does "but" means???)

Something like following code might be better.

```go
	if want := "My-Blog"; annotation.Service != want {
		t.Errorf("request json: Service=%v, want %v", annotation.Service, want)
	}
```
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

But, this is nit comment. you can ignore this.

Comment (syou6162) [inline graph_annotation_test.go]: It seems mackerel-client-go contains such error messages in several tests. I'll send another pull request for the modifications of the error message.

**Pull request merged**